src/cs/Ssh/IO/SshProtocol.cs

```diff
@@ -160,19 +163,25 @@ internal async Task<string> ReadProtocolVersionAsync(CancellationToken cancellat
 				if (line.StartsWith("SSH-", StringComparison.Ordinal))
 				{
 					this.metrics.AddMessageReceived(i + 1);
-					return line;
+					return (line, proxyIPAddress);
+				}
+				else if (line.StartsWith("PROXY ", StringComparison.Ordinal))
+				{
+					var split = line.Split(' ');
```

Comment:
Would it be useful to also get the destination port from the PROXY protocol here (`split[5]`) and expose that on the session? That would allow for more flexibility in how public port SSH numbers are mapped.

Reply:
I have done this change in tunnels repository. And will add the destination port there.
Abandoning this PR as we will not be doing this here.